commit 154feb3a4ffbf861746ace65dc62e57b107dccf7 (refs/changes/48/948/1)
Author: Richard Bradley <richard.bradley@joyent.com>
Date:   2016-11-24T09:58:54+00:00 (2 years, 10 months ago)
    
    MANTA-3036 validate.js should verify this_az exists, and is accounted for in azs and network definitions

diff --git a/networking/validate.js b/networking/validate.js
index b749a6a..bb11df9 100644
--- a/networking/validate.js
+++ b/networking/validate.js
@@ -75,6 +75,14 @@ function main()
 	if (spec['manta_nodes'].length === 0)
 		fatal('no indexing CNs listed');
 
+	if (!('this_az' in spec)) {
+		fatal('missing this availability zone');
+	}
+
+	if (!(typeof (spec['this_az']) === 'string')) {
+		fatal('this availability zone must be a string');
+	}
+
 	if (!('azs' in spec))
 		fatal('missing availability zone list');
 
@@ -84,6 +92,11 @@ function main()
 	if (spec['azs'].length === 0)
 		fatal('no azs listed');
 
+	if (spec['azs'].indexOf(spec['this_az']) < 0) {
+		fatal('this availability zone missing from ' +
+		    'availability zone list');
+	}
+
 	if (!('admin' in spec))
 		fatal('missing admin network information');
 
